website/static/js/nodesPrivacy.js

```diff
@@ -0,0 +1,281 @@
+        self.page('select');
+    };
+
+    self.confirmWarning =  function() {
+        var nodesState = ko.toJS(self.nodesState);
+        for (var node in nodesState) {
+            if (nodesState[node].changed) {
+                if (nodesState[node].public) {
+                    self.nodesChangedPublic().push(nodesState[node].title);
+                }
+                else {
+                    self.nodesChangedPrivate().push(nodesState[node].title);
+                }
+            }
+        }
+
+        self.page('confirm');
+    };
+
+    self.confirmChanges =  function() {
+        var nodesState = ko.toJS(self.nodesState());
+        //patchNodesPrivacy(nodesState);
+        nodesState = Object.keys(nodesState).map(function(key) {
+            return nodesState[key];
+        });
+        var nodesChanged = nodesState.filter(function(node) {
+            return node.changed;
+        });
+        patchNodesPrivacy(nodesChanged).then(function () {
+            window.location.reload();
+        });
+    };
+
+    self.clear = function() {
+        self.nodesChangedPublic([]);
+        self.nodesChangedPrivate([]);
+        self.page('warning');
+    };
+
+    self.selectAll = function() {
+        var nodesState = ko.toJS(self.nodesState());
+        for (var node in nodesState) {
+            if (nodesState[node].canWrite) {
+                nodesState[node].public = true;
+                if (nodesState[node].public !== self.nodesOriginal[node].public) {
+                    nodesState[node].changed = true;
+                }
+                else {
+                    nodesState[node].changed = false;
+                }
+            }
+        }
+        self.nodesState(nodesState);
+        m.redraw(true);
+    };
+
+    self.selectNone = function() {
+        var nodesState = ko.toJS(self.nodesState());
+        for (var node in nodesState) {
+            if (nodesState[node].canWrite) {
+                nodesState[node].public = false;
+                if (nodesState[node].public !== self.nodesOriginal[node].public) {
+                    nodesState[node].changed = true;
+                }
+                else {
+                    nodesState[node].changed = false;
+                }
+
+            }
+        }
+        self.nodesState(nodesState);
+        m.redraw(true);
+    };
+
+    self.back = function() {
+        var self = this;
+        self.nodesChangedPublic([]);
+        self.nodesChangedPrivate([]);
+        self.page('select');
+    };
+
+};
+
+function NodesPrivacy (selector, data, parentNodePrivacy) {
+    var self = this;
+    self.selector = selector;
+    self.$element = $(self.selector);
+    self.data = data;
+    self.viewModel = new NodesPrivacyViewModel(self.data, parentNodePrivacy);
+    self.viewModel.fetchNodeTree().done(function(response) {
+        new NodesPrivacyTreebeard('nodesPrivacyTreebeard', response, self.viewModel.nodesState, self.viewModel.nodesOriginal);
+    });
+    self.init();
```

Comment:
Minor: There's no need to have the `init` method if you're going to call it in the constructor function anyway. If you want to defer binding the viewmodel, then remove this line. Otherwise, just do the binding in this constructor.